### PR TITLE
remove resolver

### PR DESCRIPTION
**Pull request by marten-seemann**
This implements option 1 suggested in https://github.com/grandcat/zeroconf/issues/98.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +39,7 @@ go func(results <-chan *zeroconf.ServiceEntry) {
 
 ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
 defer cancel()
+// Discover all services on the network (e.g. _workstation._tcp)
 err = resolver.Browse(ctx, "_workstation._tcp", "local.", entries)
```
Comment (Stebalien) [inline README.md:43]: fixme

File: client.go [hunk 2]
```diff
@@ -52,99 +59,70 @@ func SelectIfaces(ifaces []net.Interface) ClientOption {
...
 	}
 	params.Entries = entries
-	ctx, cancel := context.WithCancel(ctx)
-	go r.c.mainloop(ctx, params)
-	err := r.c.query(params)
-	if err != nil {
-		// cancel mainloop
-		cancel()
-		return err
+	return cl.run(ctx, params)
```
Comment (Stebalien) [inline client.go:89]: This blocks now, right? Did it used to block?

Comment (marten-seemann) [reply client.go:89]: It does now, but it didn't before.

Comment (Stebalien) [reply client.go:89]: I see. We should definitely document that as users will need to read from the channel in a separate goroutine.

**Review by Stebalien — APPROVED** [review]
lgtm modulo the docs nit.

File: client.go (hunk 2 above)
Comment (marten-seemann) [reply client.go:93]: Done.

**Pull request merged**